src/Mvc/test/WebSites/RoutingWebSite/StartupRoutingDifferentBranches.cs

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.ApplicationModels;
+using Microsoft.AspNetCore.Mvc.Infrastructure;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace RoutingWebSite
+{
+    public class StartupRoutingDifferentBranches
+    {
+        // Set up application services
+        public void ConfigureServices(IServiceCollection services)
+        {
+            var pageRouteTransformerConvention = new PageRouteTransformerConvention(new SlugifyParameterTransformer());
+
+            services
+                .AddMvc(ConfigureMvcOptions)
+                .AddNewtonsoftJson()
+                .AddRazorPagesOptions(options =>
+                {
+                    options.Conventions.AddPageRoute("/PageRouteTransformer/PageWithConfiguredRoute", "/PageRouteTransformer/NewConventionRoute/{id?}");
+                    options.Conventions.AddFolderRouteModelConvention("/PageRouteTransformer", model =>
+                    {
+                        pageRouteTransformerConvention.Apply(model);
+                    });
+                })
+                .SetCompatibilityVersion(CompatibilityVersion.Latest);
+
+            ConfigureRoutingServices(services);
+
+            services.AddScoped<TestResponseGenerator>();
+            services.AddSingleton<IActionContextAccessor, ActionContextAccessor>();
```

Comment:
Is this necessary?

Reply:
Added a comment in code, but this is required for the testresponsegenerator.